machineconfiguration/v1alpha1/types_machineosconfig.go

```diff
@@ -0,0 +1,233 @@
+	// +optional
+	BaseOSExtensionsImagePullspec string `json:"baseOSExtensionsImagePullspec"`
+	// baseOSImagePullspec is the base OSImage we use to build our custom image.
+	// the MachineOSConfig object will use the in cluster image registry configuration.
+	// if you wish to use a mirror or any other settings specific to registries.conf, please specify those in the cluster wide registries.conf.
+	// The format of the image pullspec is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$'))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	BaseOSImagePullspec string `json:"baseOSImagePullspec"`
+	// baseImagePullSecret is the secret used to pull the base image.
+	// +kubebuilder:validation:Required
+	BaseImagePullSecret ImageSecretObjectReference `json:"baseImagePullSecret"`
+	// renderedImagePushSecret is the secret used to connect to a user registry.
+	// the final image push and pull secrets should be separate for security concerns. If the final image push secret is somehow exfiltrated,
+	// that gives someone the power to push images to the image repository. By comparison, if the final image pull secret gets exfiltrated,
+	// that only gives someone to pull images from the image repository. It's basically the principle of least permissions.
+	// this push secret will be used only by the MachineConfigController pod to push the image to the final destination. Not all nodes will need to push this image, most of them
+	// will only need to pull the image in order to use it.
+	// +kubebuilder:validation:Required
+	RenderedImagePushSecret ImageSecretObjectReference `json:"renderedImagePushSecret"`
+	// renderedImagePullSecret is the secret used to pull the final produced image.
+	// the final image push and pull secrets should be separate for security concerns. If the final image push secret is somehow exfiltrated,
+	// that gives someone the power to push images to the image repository. By comparison, if the final image pull secret gets exfiltrated,
+	// that only gives someone to pull images from the image repository. It's basically the principle of least permissions.
+	// this pull secret will be used on all nodes in the pool. These nodes will need to pull the final OS image and boot into it using rpm-ostree or bootc.
+	// +optional
+	RenderedImagePullSecret ImageSecretObjectReference `json:"renderedImagePullSecret"`
```

Comment:
Why doesn't this option live on the MachineConfigPool?  Wouldn't this be natural to 
1. be able to configure even if I'm not using machineosconfig
2. be able to configure even if I cannot change the inputs to the built image
3. be able to update without triggering a new machineosbuild

Reply:
Yeah, I agree with @cdoern. While it might seem more natural to add these to a MachineConfigPool, it feels less natural when someone hasn't opted into on-cluster layering since these settings would be irrelevant in that situation. Because on-cluster layering is optional, I don't think it makes sense to directly attach this to the MachineConfigPools.